File: codeWars/counting_duplicates/counting_duplicates.py
# Counting Duplicates


def count_duplicates(string):
    char_count = {}

    # Convert the string to lowercase to make the comparison case-insensitive
    lower_case_str = string.lower()

    # Count the occurrences of each character
    for char in lower_case_str:
        char_count[char] = char_count.get(char, 0) + 1

    # Count the characters that occur more than once
    duplicate_count = 0
    for count in char_count.values():
        if count > 1:
            duplicate_count += 1

    return duplicate_count


result1 = count_duplicates("abcde")
print(result1)
# Output: 0

result2 = count_duplicates("aabbcde")
print(result2)
# Output: 2

result3 = count_duplicates("aabBcde")
print(result3)
# Output: 2

File: codeWars/katas/counting_duplicates/counting_duplicates.js
// Counting Duplicates

function countDuplicates(str) {
    const charCount = {};

    // Convert the string to lowercase to make the comparison case-insensitive
    const lowerCaseStr = str.toLowerCase();

    // Count the occurrences of each character
    for (let char of lowerCaseStr) {
        charCount[char] = (charCount[char] || 0) + 1;
    }

    // Count the characters that occur more than once
    let duplicateCount = 0;
    for (let char in charCount) {
        if (charCount[char] > 1) {
            duplicateCount++;
        }
    }

    return duplicateCount;
}

const result1 = countDuplicates("abcde");
console.log(result1);
// Output: 0

const result2 = countDuplicates("aabbcde");
console.log(result2);
// Output: 2

const result3 = countDuplicates("aabBcde");
console.log(result3);
// Output: 2